packages/massa-proto-cli/src/as-gen.ts

```diff
@@ -0,0 +1,135 @@
+      throw new Error(`Unsupported type: ${type}`);
+  }
+}
+
+/**
+ * Creates imports for the caller file generated by {@link generateAsCall}
+ *
+ * @param protoData - the proto file data
+ *
+ * @returns the multiline import statements as a string
+ */
+function generateAsImports(protoData: ProtoFile): string {
+  let imports = `import { encode${protoData.funcName}Helper, ${protoData.funcName}Helper } `;
+  imports += `from './${protoData.funcName}Helper';\n`;
+
+  if (protoData.resType !== null) {
+    imports += `import { decode${protoData.funcName}RHelper, ${protoData.funcName}RHelper }`;
+    imports += `from './${protoData.funcName}RHelper';\n`;
+  }
+
+  imports += 'import { call } from "@massalabs/massa-as-sdk";\n\n';
+
+  return imports;
+}
+
+/**
+ * Creates a contract function caller with the given proto file and address.
+ *
+ * @param protoData - the proto file containing the informations to call the contract's function
+ * @param address - the address of the contract containing the function to call
+ * @param outputDirectory - the output directory to create the file for the caller
+ */
+function generateAsCall(
+  protoData: ProtoFile,
+  address: string,
+  outputDirectory: string,
+) {
+  // generating AS arguments
+  let args: string[] = [];
+  protoData.argFields.forEach(({ name, type }) =>
+    args.push(`${name}: ${convertTypeToAS(type)}`),
+  );
+
+  // generating caller
+  let caller = `export function ${protoData.funcName}(${
+    args.length > 0 ? args.join(', ') + ', coins: number' : 'coins: number'
+  }) : ${protoData.resType !== null ? protoData.resType : 'void'} {\n`;
+  caller += `  const result = call("${address}", "${
+    protoData.funcName
+  }", changetype<StaticArray<u8>>(encode${protoData.funcName}Helper(new ${
+    protoData.funcName
+  }Helper(${Array.from(protoData.argFields.keys()).join(', ')}))), coins);\n`;
+
+  if (protoData.resType !== null) {
+    caller += `  const response = decode${protoData.funcName}RHelper(`;
+    caller += `Uint8Array.wrap(changeType<ArrayBuffer>(result)));\n\n`;
+    caller += '  return response.value;\n';
+  }
+  caller += '}';
+
+  writeFileSync(
+    generateAsImports(protoData) + caller,
+    path.join(outputDirectory, protoData.funcName + '.ts'),
+  );
+}
+/**
+ * Creates the assembly script helper for serializing and deserializing with the given protobuf file.
+ *
+ * @param protoData - the proto file data.
+ * @param outputDirectory - the directory where to generate such helpers.
+ */
+function generateAsHelper(protoData: ProtoFile, outputDirectory: string) {
+  let protocProcess = spawnSync('protoc', [
+    `--plugin=protoc-gen-as=./node_modules/.bin/as-proto-gen`,
+    `--as_out=${outputDirectory}`,
+    `--as_opt=gen-helper-methods`,
+    `--proto_path=./build`,
+    protoData.protoData,
+  ]);
+
+  if (protocProcess.status !== 0) {
+    console.error(
+      `Failed to generate AS helpers code for ${protoData} with error: ${protocProcess.stderr}`,
+    );
```

Comment:
I think that @Elli610 throws instead of log error. Can you think with him? Furtehrmore, why do you continue the job here ?

Reply:
You are right, fixing that bad choice of error handeling here.